src/librustc_trans/back/write.rs

```diff
@@ -1304,6 +1365,8 @@ fn start_executing_work(tcx: TyCtxt,
         let mut compiled_modules = vec![];
         let mut compiled_metadata_module = None;
         let mut compiled_allocator_module = None;
+        let mut needs_lto = Vec::new();
```

Comment:
Would mind updating the big comment above with how LTO fits into the picture. It's not immediately clear to me what work gets executed when and where -- e.g. it seems that a whole bunch of work gets done on the scheduler thread via `generate_lto_work`. That deserves to be mentioned, I think.